okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/HostEventsSink.java

```diff
@@ -26,4 +26,25 @@
     void record(String serviceName, String hostname, int port, int statusCode, long micros);
 
     void recordIoException(String serviceName, String hostname, int port);
+
+    default HostEventCallback callback(String serviceName, String hostname, int port) {
```

Comment:
javadoc saying to prefer this version for performance reasons?

Reply:
It depends on the caller, the way okhttp is built it wouldn't make sense to construct a single-use callback, but if you have a channel representation that's reused, it's much better to use the callback approach.

Reply:
Ok sounds like that context you just shared would be helpful info for callers of this interface!